### PR TITLE
Benchmark time it takes compile `Standard.Base` library

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Benchmark time it takes to `enso --compile ... Standard/Base` library before speeding it up by [caching libraryName of ImportTarget](https://github.com/enso-org/enso/pull/11355/commits/43021fcef0506bdebc6e9f5a24a7f1a53e4dca3d).

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
- [x] New [benchmark runs](https://github.com/enso-org/enso/actions/runs/11399009701) and [gets improved](https://github.com/enso-org/enso/actions/runs/11400535211) by **10%** by the changes in this PR

### CODE REVIEW

File: test/Benchmarks/src/Compiler/Compile.enso [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from Standard.Test import Bench
+
+polyglot java import java.lang.System as Java_System
+polyglot java import java.io.File as Java_File
+
+type Data
+    Value ~enso_bin:File ~std_base:File
+
+    bench_compile_base self =
+        self.startup [ "--no-compile-dependencies", "--no-global-cache", "--compile", self.std_base.to_text]
```
Comment (Akirathan) [inline test/Benchmarks/src/Compiler/Compile.enso]: What about `--no-ir-caches`?

File: test/Benchmarks/src/Compiler/Compile.enso
Comment (Akirathan) [inline test/Benchmarks/src/Compiler/Compile.enso:1]: The benchmark looks fine in general. But just to be sure that the compilation is actually done and nothing is read from caches, I would also try to delete `.enso/cache` directory in the library. And after the benchmark is done, maybe check that these `.enso/cache` directory is present?

Comment (Akirathan) [reply test/Benchmarks/src/Compiler/Compile.enso:1]: The problem is that `std-benchmarks/bench` depend on `buildEngineDistribution`, which creates these `.enso/cache` directories. And I am not 100% convinced that just passing `--no-ir-caches` will force the compiler not to read the caches from `.enso/cache` directory. The only way to be 100% sure is to delete them explicitly. Or check the output from `--log-level INFO`?

**Comment by JaroslavTulach** [comment]
The new benchmark
- [runs](https://github.com/enso-org/enso/actions/runs/11399009701) in `11.6s`
- [gets improved](https://github.com/enso-org/enso/actions/runs/11400535211) to `10.2s`
 
e.g. there is a **10%** speedup by the changes in this PR.

File: test/Benchmarks/src/Compiler/Compile.enso
Comment (JaroslavTulach) [reply test/Benchmarks/src/Compiler/Compile.enso:1]: Checking from command line. It has to be `--log-level debug`. And yes, it is necessary to add `--no-ir-caches` as well.

Comment (JaroslavTulach) [reply test/Benchmarks/src/Compiler/Compile.enso:1]: > caches from `.enso/cache` directory. The only way to be 100% sure is to delete them explicitly.

That's a good idea. Done in c1d4dce10b and verified manually that the caches are deleted before each benchmark iteration.

**Review by Akirathan — APPROVED** [review]

File: test/Benchmarks/src/Compiler/Compile.enso (hunk 1 above)
Comment (JaroslavTulach) [reply test/Benchmarks/src/Compiler/Compile.enso]: The [default generator in sbt](https://github.com/enso-org/enso/blob/27a535f6d068f0a2be9f9825b7b5214e8f9ba5ff/project/DistributionPackage.scala#L240) isn't using them either. I want to be close to "user experience" - e.g. our developer experience.

**Review by hubertp — APPROVED** [review]

File: test/Benchmarks/src/Compiler/Compile.enso (hunk 1 above)
Comment (Akirathan) [reply test/Benchmarks/src/Compiler/Compile.enso]: Yes, but the default generator in sbt tries to be as fast as possible (so for example it does not disable IR caches). Whereas in this benchmark, you try to do a cold, reproducible, compilation.

Comment (JaroslavTulach) [reply test/Benchmarks/src/Compiler/Compile.enso]: > The [default generator in sbt](https://github.com/enso-org/enso/blob/27a535f6d068f0a2be9f9825b7b5214e8f9ba5ff/project/DistributionPackage.scala#L240) isn't using them either. I want to be close to "user experience" - e.g. our developer experience.

The major motivation for this PR is 

> generator in sbt tries to be as fast as possible

to make sure the _generator in sbt is as fast as possible_. That's why this new benchmarks mimics what the _generator in sbt_ does as closely as possible.

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**